### PR TITLE
dynamic_modules: exclude abi.h from clang-tidy

### PR DESCRIPTION
**Pull request by mathetake**
Commit Message: dynamic_modules: dynamic_modules: exclude abi.h from clang-tidy
Additional Description:

This commit disables clang-tidy for the whole ABI header which is a pure C file. 
Previously, I added `// NOLINT` comment line on each place, 
clang-tidy is only for C++ source code, so this disables it considering the 
readability. This shouldn't be a problem as the header only has types and 
function declarations.

Risk Level: none
Testing: n/a
Docs Changes: n/a
Release Notes: n/a
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Fixes commit #PR or SHA]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

**Comment by mathetake** [comment]
cc @phlax

**Comment by mathetake** [comment]
Note: I wanted to use `// NOLINTBEGIN(modernize-use-using)` rather than specifying in .clang-tidy.yaml but looks like it's available from LLVM 15 vs we are still using 14

**Comment by mathetake** [comment]
so as discussed in https://github.com/envoyproxy/envoy/pull/37320 clang-tidy is currently disabled in the CI, so the change here is not checked there... At least it's working locally ;)

File: .clang-tidy [hunk 1]
```diff
@@ -39,6 +39,11 @@ Checks: >
   readability-redundant-smartptr-get,
   readability-redundant-string-cstr
 
+Checks: '-modernize-use-using'
```
Comment (phlax) [inline .clang-tidy:42]: can the `Checks` field be repeated?

Comment (mathetake) [reply .clang-tidy:42]: good q. will take a look tomorrow!

Comment (phlax) [reply .clang-tidy:42]: probably this line isnt necessary - `modernize-use-using` is already listed i think

Comment (phlax) [reply .clang-tidy:42]: (or just needs to be removed above perhaps)

**Comment by mathetake** [comment]
@phlax i went ahead and simply put the header in the ignore target of clang-tidy - see the description of the PR for detail!

**Review by phlax — APPROVED** [review]
lgtm, thanks @mathetake

**Pull request merged**